src/MqttHandleInverter.cpp

```diff
@@ -247,4 +253,17 @@ void MqttHandleInverterClass::onMqttMessage(const espMqttClientTypes::MessagePro
             MessageOutput.println("Ignored because retained");
         }
     }
+}
+
+String MqttHandleInverterClass::getSubTopic(std::shared_ptr<InverterAbstract> inv)
+{
+    String subtopic;
+
+    if (MqttSettings.getSubTopicInverter()) {
+        subtopic = inv->name();
```

Comment:
I think you must take care of some special characters here: `+`, `#` and `*` are wildcard characters and not allowed for publishing.  Also the `$` character is not allowed for mqtt clients. If someone enters a space in the inverter name it would break the home assistent auto discovery. Also slashes (`/`) must not occour in the inverter name. 

So somehow these characters have to be replaced or removed from the topic. I could imagine to replace them by a underscore (`_`)

Reply:
Good point. Will add a helper function to replace such characters.